storage/backend/batch_tx.go

```diff
@@ -0,0 +1,98 @@
+package backend
+
+import (
+	"bytes"
+	"log"
+	"sync"
+
+	"github.com/boltdb/bolt"
+)
+
+type BatchTx interface {
+	Lock()
+	Unlock()
+	UnsafeCreateBucket(name []byte)
+	UnsafePut(bucketName []byte, key []byte, value []byte)
+	UnsafeRange(bucketName []byte, key, endKey []byte, limit int64) [][]byte
+	UnsafeDelete(bucketName []byte, key []byte)
+}
+
+type batchTx struct {
+	mu      sync.Mutex
```

Comment:
> Should probably be directly embedded 

I do not quite like embedded thing in a private struct. The embedded field is always public accessible. 
In this case, people can access batchTx.Mutex. But this is probably OK here.